docs/creating-a-runtime-check.md

```diff
@@ -0,0 +1,209 @@
+
+### Check Preperations
+
+Check preparations are created using the `prepare()` method within a check class.
+
+This type of preparation is useful for carrying out the logic to prepare the environment that is specific to the check that implements it.
+
+Below is an example of a preparation that creates a test post and returns a cleanup function that deletes that post after the check is run.
+
+```php
+use WordPress\Plugin_Check\Checker\Checks\Abstract_Runtime_Check;
+
+class Custom_Check extends Abstract_Runtime_Check {
+  /**
+   * Check Preparation.
+   */
+  public function prepare() {
+    // Create a test post ahead of running the check.
+    $post_id = wp_insert_post(
+      array(
+        'post_title'   => 'Test post',
+        'post_content' => 'This is a test post.',
+        'post_type'    => 'post',
+      )
+    );
+
+    // Return the clean up function.
+    return function() use ( $post_id ) {
+      // Remove the test post created.
+      wp_delete_post( $post_id );
+    }
+  }
+}
+```
+
+### Preparation Classes
+
+Preparation classes are individual classes created to handle specific preparation logic in a single place.
+
+The [`Force_Single_Plugin_Preparation`](https://github.com/10up/plugin-check/blob/trunk/includes/Checker/Preparations/Force_Single_Plugin_Preparation.php) in the Plugin Checker is an example of a preparation class.
+
+Preparation classes are useful for implementing the same preparation logic across multiple checks using the concept of shared preparations.
+
+### Shared Preparations
+
+Shared preparations are used to prevent running the same preparations multiple times.
+
+Before running checks against a plugin all shared preparations are collected and processed to remove any shared preparations that are the same. 
+
+Check classes can define the shared preparations the use by using the `With_Shared_Preparation` interface.
+
+The Check class should then implement the get_shared_preparations() method defined by the interface. This method returns an map of shared preparations where the preparation class name is the key and an array of constructor parameters as the value.
+
+Below is an example of how the `Enqueued_Scripts_Size_Check` uses shared preparations.
+
+```php
+  /**
+   * Returns an array of shared preparations for the check.
+   *
+   * @return array Returns a map of $class_name => $constructor_args pairs. If the class does not
+   *               need any constructor arguments, it would just be an empty array.
+   */
+  public function get_shared_preparations() {
+    $demo_posts = array_map(
+      function( $post_type ) {
+        return array(
+          'post_title'   => "Demo {$post_type} post",
+          'post_content' => 'Test content',
+          'post_type'    => $post_type,
+          'post_status'  => 'publish',
+        );
+      },
+      $this->viewable_post_types
+    );
+
+    return array(
+      Demo_Posts_Creation_Preparation::class => array( $demo_posts ),
+    );
+  }
+```
+
```

Comment:
We need to add another section here for "Global preparations". This only needs to be an explanation that the Plugin Checker also includes some preparation classes that are used to prepare the overall environment, i.e. before _any_ runtime check. Global preparations cannot be controlled outside of the Plugin Checker at this point. An example for a global preparation is the `Force_Single_Plugin_Preparation` class, which ensures that only the plugin to check is actually active.